pkg/stanza/fileconsumer/config.go

```diff
@@ -139,3 +129,54 @@ func (c Config) Build(logger *zap.SugaredLogger, emit EmitFunc) (*Manager, error
 		seenPaths:     make(map[string]struct{}, 100),
 	}, nil
 }
+
+func (c Config) validate() error {
+	if len(c.Include) == 0 {
+		return fmt.Errorf("required argument `include` is empty")
+	}
+
+	// Ensure includes can be parsed as globs
+	for _, include := range c.Include {
+		_, err := doublestar.PathMatch(include, "matchstring")
+		if err != nil {
+			return fmt.Errorf("parse include glob: %w", err)
+		}
+	}
+
+	// Ensure excludes can be parsed as globs
+	for _, exclude := range c.Exclude {
+		_, err := doublestar.PathMatch(exclude, "matchstring")
+		if err != nil {
+			return fmt.Errorf("parse exclude glob: %w", err)
+		}
+	}
+
+	if c.MaxLogSize <= 0 {
+		return fmt.Errorf("`max_log_size` must be positive")
+	}
+
+	if c.MaxConcurrentFiles <= 1 {
+		return fmt.Errorf("`max_concurrent_files` must be greater than 1")
+	}
+
+	if c.FingerprintSize < MinFingerprintSize {
+		return fmt.Errorf("`fingerprint_size` must be at least %d bytes", MinFingerprintSize)
+	}
+
+	_, err := c.Splitter.EncodingConfig.Build()
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func judgeStartAtBeginning(startAt string) (bool, error) {
+	switch startAt {
+	case "beginning":
+		return true, nil
+	case "end":
+		return false, nil
+	default:
+		return false, fmt.Errorf("invalid start_at location '%s'", startAt)
+	}
+}
```

Comment:
nit: Since we only use this once, I think we can just move the code into `buildManager`

Reply:
This function is already used in buildManager.  We can see this function on line 103.